xcode/Subconscious/Shared/Components/FirstRun/FirstRunView.swift

```diff
@@ -46,26 +46,62 @@ struct FirstRunView: View {
                 
                 Spacer()
                 
-                ValidatedFormField(
-                    alignment: .center,
-                    placeholder: "Enter your invite code",
-                    field: app.state.inviteCodeFormField,
-                    send: Address.forward(
-                        send: app.send,
-                        tag: AppAction.inviteCodeFormField
-                    ),
-                    caption: "Look for this in your welcome email.",
-                    submitLabel: .go,
-                    onSubmit: {
-                        if app.state.inviteCodeFormField.isValid {
-                            app.send(.pushFirstRunStep(.nickname))
+                if let gatewayId = app.state.gatewayId {
+                    VStack(spacing: AppTheme.unit2) {
+                        HStack(spacing: AppTheme.unit2) {
+                            Text("Invite code redeemed")
+                        Image(systemName: "checkmark.circle")
+                            .resizable()
+                            .frame(width: 16, height: 16)
+                            .opacity(0.5)
                         }
+                        .font(.body)
+                        .bold()
+                        
+                        HStack(spacing: AppTheme.unit) {
+                            Text("Gateway ID")
+                                .bold()
+                            Text(gatewayId)
+                        }
+                        .font(.caption)
                     }
-                )
-                .textFieldStyle(.roundedBorder)
-                .textInputAutocapitalization(.never)
-                .disableAutocorrection(true)
-                
+                    .foregroundColor(.secondary)
+                } else {
+                    ValidatedFormField(
+                        alignment: .center,
+                        placeholder: "Enter your invite code",
+                        field: app.state.inviteCodeFormField,
+                        send: Address.forward(
+                            send: app.send,
+                            tag: AppAction.inviteCodeFormField
+                        ),
+                        caption: Func.run {
+                            switch app.state.gatewayProvisioningStatus {
+                            case .failed(_):
+                                return "Failed to redeem invite code"
+                            case _:
+                                return "Look for this in your welcome email"
+                            }
```

Comment:
Display error state as a validation error. This meant expanding the API of `ValidatedFormField` but I also think it's the best way to handle this error?

<img width="358" alt="image" src="https://github.com/subconsciousnetwork/subconscious/assets/5009316/512b2249-7087-4f33-85ae-cbb734c8aadb">

Reply:
Small tweak request: "Could not redeem invite code".

We use "succeed" / "fail" under the hood, mirroring the terminology used for the Result type. However, I think "fail" is a scary word to present to users.